lib/Optimizer/GraphOptimizer.cpp

```diff
@@ -909,6 +985,45 @@ static NodeValue simplifyConcatNode(Function *F, ConcatNode *CN) {
     }
   }
 
+  /// If the inputs of concat are all reshape nodes, and the inputs of reshape
+  /// are able to be concatenated, then we can perform concat first and then
+  /// reshape. Concat(Reshape(x) * N) -> Reshape(Concat(x * N)).
+  {
+    std::vector<ReshapeNode *> reshapes;
+    bool optimizable = true;
+    for (auto &I : CN->getInputs()) {
+      if (auto *S = dyn_cast<ReshapeNode>(I)) {
```

Comment:
may be: auto *reshape or auto *R. What does "S" mean here?

Reply:
changed to R